config/environments/production.rb

```diff
@@ -12,6 +14,18 @@
   # Rake tasks automatically ignore this option for performance.
   config.eager_load = true
 
+  s3_options = {
+    access_key_id: Rails.application.credentials.dig(:aws, :access_key_id),
+    secret_access_key: Rails.application.credentials.dig(:aws, :secret_access_key),
+    region: Rails.application.credentials.dig(:aws, :region),
+    bucket: Rails.application.credentials.dig(:aws, :bucket)
+  }
+
+  Shrine.storages = {
+    cache: Shrine::Storage::S3.new(prefix: 'cache', **s3_options),
```

Comment:
Maybe `Rails.application.credentials.aws` will be enough instead of fetching each key?